File: Prometheus/Model/EvalNodeTest.cs
using System;
using NUnit.Framework;

namespace Prometheus.Model
{
	[TestFixture]
	public class EvalNodeTest
	{
		[Test]
		public void numberCheckTest()
		{
			EvalNode node = new EvalNode("Test", null){Eval = "122.1"};
			Assert.IsTrue(node.isNumber());

			node.Eval = "122";
			Assert.IsTrue(node.isNumber());

			node.Eval = ".10";
			Assert.IsTrue(node.isNumber());

			node.Eval = "10.1234";
			Assert.IsTrue(node.isNumber());

			node.Eval = "100p";
			Assert.IsFalse(node.isNumber());

			node.Eval = "x100";
			Assert.IsFalse(node.isNumber());

			node.Eval = "ethan";
			Assert.IsFalse(node.isNumber());
		}
	}
}

File: Prometheus++/Tests/EvalNodeTest.cpp
#include "TestRunner.h"
#include "../Dev/Model/EvalNode.h"
#include <string>
using namespace std;
using namespace Model;

class EvalNodeTest {
public:
  static void isNumberTest() {
	EvalNode node("Test Eval Node", 0);
	
	node.setEvaluation("1.55");
	TestRunner::assertTrue("Failure: 1.55 is a number", node.isNumber());
	
	node.setEvaluation("0.05");
	TestRunner::assertTrue("Failure: 0.05 is a number", node.isNumber());
	
	node.setEvaluation("0.0.0.0");
	TestRunner::assertFalse("Failure: 0.0.0.0 isn't a number.", node.isNumber());
	
	node.setEvaluation("oh hai");
	TestRunner::assertFalse("Failure: 'oh hai' isn't a number.", node.isNumber());
  }
};

int main() {
  TestRunner run;
  run.addTest(EvalNodeTest::isNumberTest, "isNumberTest");
  run.runTests();
  
  return 0;
}